File: Python/binarySearch.py
def binarySearch(A,key):
    hi = len(A)-1
    low = 0
    while low<= hi:
        mid = low + (hi-low)//2
        if key <A[mid]:
            hi = mid - 1
        elif key>A[mid]:
            low = mid+1
        else:
            return mid
    return -1

A = [0,1,2,3,4,5,6,7,8,9,10]

print(binarySearch(A,3))

File: C++/binarySearch.cpp
#include <iostream>

using namespace std;

int binarySearch(int A[],int N, int key)
{
    int hi = N - 1;
    int low = 0;
    while (low <= hi)
    {
        int mid = low + (hi - low) / 2;
        if (key < A[mid])
            hi = mid - 1;
        else if (key > A[mid])
            low = mid + 1;
        else
            return mid;
    }
    return -1;
}
